pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -213,6 +214,13 @@ public List<StarTreeV2> getStarTrees() {
     return _starTreeIndexContainer != null ? _starTreeIndexContainer.getStarTrees() : null;
   }
 
+  @Nullable
+  public void setValidDocIds(ImmutableRoaringBitmap validDocIds) {
+    for (int docId: validDocIds.toArray()) {
+      _validDocIds.add(docId);
```

Comment:
this is not set, but add?

Reply:
Thanks @yupeng9 

Previously i tried to set the validDocIndex directly from snapshot. so i iterate validDocId from immutableRoaringBitmap to ThreadSafeRoaringBitmap. I synced with Jackie and learned this is not correct, so I fix it in `[this commit](https://github.com/apache/pinot/pull/9062/commits/56fb2a6b629473f5a64eb945597092854a39c3ba)`